java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -1035,11 +1037,15 @@ private static List<OrcProto.StripeStatistics> deserializeStripeStats(
       long offset,
       int length,
       InStream.StreamOptions options) throws IOException {
-    InStream stream = InStream.create("stripe stats", tailBuffer, offset,
-        length, options);
-    OrcProto.Metadata meta = OrcProto.Metadata.parseFrom(
-        InStream.createCodedInputStream(stream));
-    return meta.getStripeStatsList();
+    try (InStream stream = InStream.create("stripe stats", tailBuffer, offset,
+        length, options)) {
+      OrcProto.Metadata meta = OrcProto.Metadata.parseFrom(
+          InStream.createCodedInputStream(stream));
+      return meta.getStripeStatsList();
+    } catch (InvalidProtocolBufferException e) {
+      LOG.warn("Failed to parse stripe statistics; check ORC-1361 for more details.", e);
```

Comment:
This message could be wrong because `InvalidProtocolBufferException` can happen in another cases, right?

Reply:
+1 
From [protobuf](https://github.com/protocolbuffers/protobuf/blob/520c601c99012101c816b6ccc89e8d6fc28fdbb8/java/core/src/main/java/com/google/protobuf/InvalidProtocolBufferException.java), there are many cases here.

Reply:
The first part of the message (`Failed to parse stripe statistics`) is accurate. The second part indeed can be misleading in some cases.

Instead of pointing to ORC-1361 what do you think of creating a page/section in the ORC website and document there whatever is needed? Then the message can contain the URL to this page.

Reply:
Does the exception message provide detail information? It yes, we may not need to state `check ORC-1361 for more details` explicitly.

Reply:
Maybe we just need to change the title to catch `InvalidProtocolBufferException` and let the program continue to run.
Reading large bars of statistical information is one of the causes of the exception. But throw any InvalidProtocolBufferException and this pr will allow the program to continue to run.

Reply:
We can keep running, but need to let the user know that we skipped these statistics. This can be achieved by catching exceptions, or requiring the user to configure a parameter such as skipCheckStatistics

Reply:
The details in the exception are probably enough for the user to understand that the stats are big/corrupted. What may be more difficult to understand is why the stats are big or why they were corrupted and what can they do about it. 

Providing some tips like check the strip size, number of columns, size of string stats, total size of the file, etc., should be useful and that's what I was thinking that could go in ORC-1361 or another resource.

I am fine with whatever you decide but I would strongly suggest to LOG at least the exception that was caught.

Let me know what you prefer and I will do the appropriate changes.

Reply:
It seems the content of the logging message is somewhat blocking this PR from being merged. I think this is a rather minor issue so how about keeping it dead simple (`Failed to parse stripe statistics`) and move the PR forward? WDYT?

Reply:
> It seems the content of the logging message is somewhat blocking this PR from being merged. I think this is a rather minor issue so how about keeping it dead simple (`Failed to parse stripe statistics`) and move the PR forward? WDYT?

I'm +1 on this. Thanks @zabetak

Reply:
Simplified message in https://github.com/apache/orc/pull/1402/commits/55eebc6c6b779e6063528b3514670b33d46756cf